src/Sparrow/Json/PerCoreStatic.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using Sparrow.Utils;
+
+namespace Sparrow.Json
+{
+    public sealed class PerCoreStatic<T> where T : class
+    {
+        private readonly T[] _perCoreArrays;
+
+        public PerCoreStatic(Func<T> creationFunc)
+        {
+            _perCoreArrays = new T[Environment.ProcessorCount];
+            for (int i = 0; i < _perCoreArrays.Length; i++)
+                _perCoreArrays[i] = creationFunc();
+        }
+
+        public T Get()
+        {
+            return _perCoreArrays[CurrentProcessorIdHelper.GetCurrentProcessorId() % _perCoreArrays.Length];
```

Comment:
Don't we need to worry about false sharing here?
Shouldn't we have something like:
```
private struct Item
{
   public T Actual
   public fixed byte Padding[64];
}
```

Or something like this?

Reply:
Then again, we are never _writing_ to it, right?
So there is no sharing at all, just concurrent reads

Reply:
As we don't write, the answer is no. It would be though if we would be writing to it.